app/models/company.rb

```diff
@@ -10,8 +12,32 @@ class Company < ApplicationRecord
 
   has_many :business_categories, through: :membership_applications
 
-  has_many :membership_applications
+  has_many :membership_applications, dependent: :destroy, inverse_of: :company
 
   belongs_to :region
 
+
+  def destroy_checks
+
+    error_if_has_accepted_applications?
+
+  end
+
+
+  # do not delete a Company if it has MembershipApplications that are accepted
+  def error_if_has_accepted_applications?
+
+    membership_applications.reload
+    business_categories.reload
```

Comment:
I suspect that's because you're using an attribute reference (```admin```) rather than a method (```admin?```).  No change needed here - just a matter of style and preference, I guess.